forum/qa-plugin/user-activity-log/user-activity-log-search.php

```diff
@@ -154,28 +152,9 @@ private function validateDate(string $finalQuery)
     {
         $isValid = true;
 
-        if(!empty($this->searchArr['date'])){
-            $dateArr = explode('-', $this->searchArr['date']);
-            if(isset($dateArr[0]) && is_numeric($dateArr[0])){
-                $date = $dateArr[0];
-                if(isset($dateArr[1]) && is_numeric($dateArr[1])){
-                    $date = $date.'-'.$dateArr[1];
-                    if(isset($dateArr[2])){
-                        if(strlen($dateArr[2]) > 2){
-                            [$day, $hours] = explode(' ', $dateArr[2]);
-                            $dateArr[2] = $day;
-                            $dateArr[3] = $hours;
-                        }
-
-                        $date = $date.'-'.$dateArr[2];
-                    }
-                }
-            }else{
-                $isValid = false;
-            } 
-        }else{
+        if(!preg_match("/^(?<year>\d{4})(?:-(?<month>\d{2})(?:-(?<day>\d{2})(?:[ T](?<hour>\d{2})(?::(?<min>\d{2})?)(?::(?<sec>\d{2}))?)?)?)?$/", $this->searchArr['date'])){
```

Comment:
Wg mnie czytelniej by było wrzucić do zmiennej wyrażenie z wewnątrz nagłówka `if` - wtedy nie trzeba przyglądać się o co chodzi, tylko sama nazwa zmiennej wstępnie to zasugeruje, np.:
```php
$isDateCorrect = preg_match("/^(?<year>\d{4})(?:-(?<month>\d{2})(?:-(?<day>\d{2})(?:[ T](?<hour>\d{2})(?::(?<min>\d{2})?)(?::(?<sec>\d{2}))?)?)?)?$/", $this->searchArr['date']);

if (!$isDateCorrect) {
```